src/clib/pioc.c

```diff
@@ -973,7 +973,7 @@ int PIOc_init_decomp(int iosysid, int pio_type, int ndims, const int *gdimlen, i
                      const PIO_Offset *compmap, int *ioidp, int rearranger,
                      const PIO_Offset *iostart, const PIO_Offset *iocount)
 {
-    PIO_Offset compmap_1_based[maplen];
+    PIO_Offset* compmap_1_based = (PIO_Offset*)malloc(maplen * sizeof(PIO_Offset));
```

Comment:
@dqwu. please also add a check for malloc() fail in a separate commit before merging these changes.

Reply:
OK, I have added that check in a separate commit.